r/R/dataset-write.R

```diff
@@ -71,25 +72,30 @@
 #'  d <- dplyr::group_by(mtcars, cyl, gear)
 #'
 #'  d %>%
-#'   write_dataset(two_part_dir_2, "parquet")
+#'   write_dataset(two_part_dir_2)
 #'
 #'  # Just passing an additional hive_style option to see the difference
 #'  # in the output
 #'
 #'  d %>%
-#'   write_dataset(two_part_dir_2, "parquet", hive_style = FALSE)
+#'   write_dataset(two_part_dir_2, hive_style = FALSE)
 #'
 #'  list.files(two_part_dir_2, pattern = "parquet", recursive = TRUE)
 #'  list.files(two_part_dir_3, pattern = "parquet", recursive = TRUE)
 #' }
 #' @export
 write_dataset <- function(dataset,
                           path,
-                          format = dataset$format,
+                          format = "parquet",
```

Comment:
I don't think this is the right change. Please see the proposal in https://issues.apache.org/jira/browse/ARROW-11582?focusedCommentId=17292994&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17292994. I'd recommend reverting this change and doing the format argument change in a separate PR associated with that JIRA issue, rather than grow the scope of this one.